src/explore-education-statistics-frontend/src/modules/table-tool/components/TableToolShare.tsx

```diff
@@ -34,17 +38,26 @@ const TableToolShare = ({ tableHeaders, query }: Props) => {
     }
     setPermalinkLoading(true);
 
-    const { id } = await permalinkSnapshotService.createPermalink({
-      query,
-      configuration: {
-        tableHeaders: mapUnmappedTableHeaders(tableHeaders),
-      },
-    });
-
-    setPermalinkUrl(`${process.env.PUBLIC_URL}data-tables/permalink/${id}`);
-    setPermalinkLoading(false);
-
-    setScreenReaderMessage(`Shareable link generated. ${linkInstructions}`);
+    try {
+      const { id } = await permalinkSnapshotService.createPermalink({
+        query,
+        configuration: {
+          tableHeaders: mapUnmappedTableHeaders(tableHeaders),
+        },
+      });
+
+      setPermalinkUrl(`${process.env.PUBLIC_URL}data-tables/permalink/${id}`);
+      setPermalinkLoading(false);
+
+      setScreenReaderMessage(`Shareable link generated. ${linkInstructions}`);
+    } catch (err) {
+      logger.error(err);
+      togglePermalinkError();
```

Comment:
Would suggest using normal state for the permalink error so we can make the error message consistent e.g.

```tsx
const [permalinkError, setPermalinkError] = useState<string>();

// Later...

const errorMessage = 'There was a problem generating the share link.'

setPermalinkError(errorMessage);
setScreenReaderMessage(`Error: ${errorMessage}`);

// Then in the render...

{permalinkError ? (
  <ErrorMessage>
    {permalinkError}
  </ErrorMessage>
) : (
  <ButtonText onClick={handlePermalinkClick}>
    Generate shareable link
  </ButtonText>
)}
```

Reply:
I've made this change now, thanks.